### PR TITLE
AsyncMode -> ConcurrentMode

### PR DESCRIPTION
**Pull request by trueadm**
Follow up to https://github.com/facebook/react/pull/13732. Renames `AsyncMode` to `ConcurrentMode`.

### CODE REVIEW

**Comment by trueadm** [comment]
@bvaughn I thought that was the point here? It's an unstable API so can break in any version, right? I'm happy to add support in on the devtools, but the PR to rename AsyncMode has already landed in React – did you want me to revert it?

**Comment by bvaughn** [comment]
> @bvaughn I thought that was the point here? It's an unstable API so can break in any version, right? I'm happy to add support in on the devtools, but the PR to rename AsyncMode has already landed in React – did you want me to revert it?

Well, yes– the API can change in that we can e.g. rename the export with a new release of `react` _but_ we don't want a DevTools update to break compatibility with older releases. (DevTools tries to be compatible with all previous React versions.)

So in this case, we'll need the 16.x DevTools adapter to handle both `AsyncMode` and `ConcurrentMode` types. 😄

File: backend/attachRendererFiber.js [hunk 1]
```diff
@@ -71,8 +71,9 @@ function getInternalReactConstants(version) {
     };
   }
   ReactSymbols = {
-    ASYNC_MODE_NUMBER: 0xeacf,
-    ASYNC_MODE_SYMBOL_STRING: 'Symbol(react.async_mode)',
+    CONCURRENT_MODE_NUMBER: 0xeacf,
+    CONCURRENT_MODE_SYMBOL_STRING: 'Symbol(react.concurrent_mode)',
+    DEPRECATED_ASYNC_MODE_SYMBOL_STRING: 'Symbol(react.async_mode)',
```
Comment (bvaughn) [inline backend/attachRendererFiber.js]: This isn't sufficient. We need the number and string.

Comment (trueadm) [reply backend/attachRendererFiber.js]: The number is the same though?

Comment (bvaughn) [reply backend/attachRendererFiber.js]: Haha, d'oh! Ok sorry.

**Comment by probablyup** [comment]
It'd be cool if you reviewed my PR from 2 weeks ago that has its corresponding change already in react core and released. Thanks.

**Pull request merged**